### ISSUE TITLE
Async Generator Class Method not working

### ISSUE THREAD

**Issue by VIMGAS** [alangpierce/sucrase#478]
This generator method:
```js
class BlobStorage {
  ...

  /**
   * @returns {object[]}
   */
  async *listBlobs() {
    const iter = await this.containerClient().listBlobsFlat();
    return yield* iter;
  }
}
```

Is outputting a error when called:
```js
P:\xxx\xxx\xxx\xxx\xxx\xxx\src\app\services\blob\index.js:190
    const iter = await this.containerClient().listBlobsFlat();
                 ^^^^^

SyntaxError: await is only valid in async function
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Module._compile (P:\xxx\xxx\xxx\xxx\node_modules\pirates\lib\index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.newLoader [as .js] (P:\xxx\xxx\xxx\xxx\node_modules\pirates\lib\index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (P:\xxx\xxx\xxx\xxx\service\document\src\index.js:3:13)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Module._compile (P:\xxx\xxx\xxx\xxx\node_modules\pirates\lib\index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.newLoader [as .js] (P:\xxx\xxx\xxx\xxx\node_modules\pirates\lib\index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
[nodemon] app crashed - waiting for file changes before starting...
```

When transpilled it removes both the async tag and the * from the method name:
```js
class BlobStorage {
  ...

  /**
   * @returns {object[]}
   */
  listBlobs() {
    const iter = await this.containerClient().listBlobsFlat();
    return yield* iter;
  }
}
```

**Comment by VIMGAS** [comment]
Just seen Pull Request #463 already fixes this issue. Is there any time reference for when it should be released to npm?

**Comment by alangpierce** [comment]
@VIMGAS sorry for the delay in publishing! I just published Sucrase 3.11.0, which has the fix.
